test/CadlRanchProjects/inheritance/Generated/DiscriminatedClient.cs

```diff
@@ -0,0 +1,601 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Models.Inheritance
+{
+    // Data plane generated client. The Discriminated service client.
+    /// <summary> The Discriminated service client. </summary>
+    public partial class DiscriminatedClient
+    {
+        private readonly HttpPipeline _pipeline;
+        private readonly Uri _endpoint;
+        private readonly string _apiVersion;
+
+        /// <summary> The ClientDiagnostics is used to provide tracing support for the client library. </summary>
+        internal ClientDiagnostics ClientDiagnostics { get; }
+
+        /// <summary> The HTTP pipeline for sending and receiving REST requests and responses. </summary>
+        public virtual HttpPipeline Pipeline => _pipeline;
+
+        /// <summary> Initializes a new instance of DiscriminatedClient. </summary>
+        public DiscriminatedClient() : this(new Uri("http://localhost:3000"), new ModelsInheritanceClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of DiscriminatedClient. </summary>
+        /// <param name="endpoint"> TestServer endpoint. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> is null. </exception>
+        public DiscriminatedClient(Uri endpoint, ModelsInheritanceClientOptions options)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            options ??= new ModelsInheritanceClientOptions();
+
+            ClientDiagnostics = new ClientDiagnostics(options, true);
+            _pipeline = HttpPipelineBuilder.Build(options, Array.Empty<HttpPipelinePolicy>(), Array.Empty<HttpPipelinePolicy>(), new ResponseClassifier());
+            _endpoint = endpoint;
+            _apiVersion = options.Version;
+        }
+
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response<Fish>> GetModelValueAsync(CancellationToken cancellationToken = default)
+        {
+            using var scope = ClientDiagnostics.CreateScope("DiscriminatedClient.GetModelValue");
+            scope.Start();
+            try
+            {
+                RequestContext context = FromCancellationToken(cancellationToken);
+                Response response = await GetModelAsync(context).ConfigureAwait(false);
+                return Response.FromValue(Fish.FromResponse(response), response);
```

Comment:
Not quite understand this, will this call retun unknownFish and UnknownFish is internal.
Could you help to explain the logic here? thanks